src/authorization_token_contract/src/main.nr

```diff
@@ -0,0 +1,502 @@
+mod types;
+
+use dep::aztec::macros::aztec;
+
+// Minimal authorization token implementation that supports `AuthWit` accounts.
+// The auth message follows a similar pattern to the cross-chain message and includes a designated caller.
+// The designated caller is ALWAYS used here, and not based on a flag as cross-chain.
+// message hash = H([caller, contract, selector, ...args])
+// To be read as `caller` calls function at `contract` defined by `selector` with `args`
+// Including a nonce in the message hash ensures that the message can only be used once.
+#[aztec]
+pub contract AuthorizationToken {
```

Comment:
imho, there's not a real need of having this contract right now, we could do tests only on the Authorizer / Verifier as unit tests instead of integrating it with a token contract

helps defining the flows we need to cover tho:
- private to private
- private to public
- public to private (private!)
- send hiding point to public?
- public to public
- public to hiding point